### PR TITLE
add query string

### PR DESCRIPTION
**Pull request by domejiraphon**

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -172,11 +172,34 @@ def get_customer_addresses(customer_id):
...
+    all_query_key = ["username", "first_name", "last_name", "addresses"]
+    for key in request.args.keys():
+      if key not in all_query_key:
+        message = {
+            "error": "Conflict",
+            "message": "The query key: '" + key + "' is not supported."
+            }
 
+        return make_response(
+            jsonify(message), status.HTTP_409_CONFLICT
```
Comment (ryanjcohen) [inline service/routes.py:185]: I'm wondering if this should be a 400, but have not found a solid source indicating what to do in this case yet

File: service/routes.py [hunk 2]
```diff
@@ -172,11 +172,34 @@ def get_customer_addresses(customer_id):
...
+    last_name = request.args.get("last_name")
+    #addresses = request.args.get("addresses")
+    if username:
+      customers = Customer.find_by_name(username)
+    elif first_name:
+      customers = Customer.find_by_first_name(first_name)
+    elif last_name:
+      customers = Customer.find_by_last_name(last_name)
+    #elif addresses:
+    #  customers = Customer.find_by_addresses(addresses)
```
Comment (ryanjcohen) [inline service/routes.py:199]: I think it's fine to not allow querying by addresses here, but I guess we should remove addresses from all_query_key on line 176 if this is the case

File: service/routes.py [hunk 3]
```diff
@@ -172,11 +172,34 @@ def get_customer_addresses(customer_id):
...
+    username = request.args.get("username")
+    first_name = request.args.get("first_name")
+    last_name = request.args.get("last_name")
+    #addresses = request.args.get("addresses")
+    if username:
+      customers = Customer.find_by_name(username)
+    elif first_name:
+      customers = Customer.find_by_first_name(first_name)
+    elif last_name:
+      customers = Customer.find_by_last_name(last_name)
```
Comment (ryanjcohen) [inline service/routes.py:197]: One thing that I'm wondering if we should support is when a user enters multiple query key/parameter pairs - ex. if a user is looking for everyone with a first name of Joe _and_ a last name of Smith

Comment (domejiraphon) [reply service/routes.py]: Fixed

File: service/routes.py (hunk 2 above)
Comment (domejiraphon) [reply service/routes.py:221]: Fixed. I already removed it.

File: service/routes.py (hunk 1 above)
Comment (domejiraphon) [reply service/routes.py]: Fixed.

**Review by ryanjcohen — APPROVED** [review]

**Pull request merged**